src/genai/_utils/general.py

```diff
@@ -152,8 +153,12 @@ def merge_objects(*objs: Optional[Mapping[_KT, _VT]]) -> dict[_KT, _VT]:
     return result
 
 
-def cast_list(input: Union[TInput, list[TInput]]) -> list[TInput]:
-    return input if isinstance(input, list) else [input]
+def is_sequence(value: Any) -> bool:
+    return False if isinstance(value, str) else isinstance(value, Sequence)
```

Comment:
why not just?

`return isinstance(value, Sequence)`

Reply:
Strings is also a Sequence, which breaks other things
maybe better name could be `is_non_string_sequence`

Reply:
or `is_list_like_sequence`

Reply:
`return isinstance(value, Sequence) and not isinstance(value, str)`